src/chi-vue/src/components/drawer/drawer.tsx

```diff
@@ -0,0 +1,213 @@
+  transitioningClass
+} from '../../constants/classes';
+import {
+  chiDrawerShowEvent,
+  chiDrawerShownEvent,
+  chiDrawerHideEvent,
+  chiDrawerHiddenEvent
+} from '../../constants/events';
+import { ThreeStepsAnimation } from '../../utils/ThreeStepsAnimation';
+import { ANIMATION_DURATION } from '../../constants/constants';
+import { contains } from '../../../../custom-elements/src/utils/utils';
+
+@Component
+export default class Drawer extends Vue {
+  @Prop() position!: "left" | "top" | "right" | "bottom";
+  @Prop() backdrop!: string;
+  @Prop() active!: boolean;
+  @Prop() nonClosable!: boolean;
+  @Prop() noHeader!: boolean;
+  @Prop() preventAutoHide!: boolean;
+  @Prop() portal!: boolean;
+  @Prop() title!: string;
+
+  animation!: ThreeStepsAnimation;
+
+  _animationClasses!: string;
+
+  _backdropAnimationClasses!: string;
+
+  show() {
+    if (this.animation && !this.animation.isStopped()) {
+      this.animation.stop();
+    }
+
+    if (this._backdropAnimationClasses !== '' || this._animationClasses !== activeClass) {
+      this.animation = ThreeStepsAnimation.animationFactory(
+        () => {
+          this._animationClasses = transitioningClass;
+          this._backdropAnimationClasses = `${transitioningClass} ${closedClass}`;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = `${transitioningClass} ${activeClass}`;
+          this._backdropAnimationClasses = transitioningClass;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = activeClass;
+          this._backdropAnimationClasses = '';
+          this.$forceUpdate();
+          this.$emit(chiDrawerShownEvent);
+        },
+        ANIMATION_DURATION.MEDIUM
+      );
+    }
+
+    this.$emit(chiDrawerShowEvent);
+  }
+
+  hide() {
+    if (this.animation && !this.animation.isStopped()) {
+      this.animation.stop();
+    }
+
+    if (this._backdropAnimationClasses !== closedClass || this._animationClasses !== '') {
+      this.animation = ThreeStepsAnimation.animationFactory(
+        () => {
+          this._animationClasses = `${transitioningClass} ${activeClass}`;
+          this._backdropAnimationClasses = transitioningClass;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = transitioningClass;
+          this._backdropAnimationClasses = `${transitioningClass} ${closedClass}`;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = '';
+          this._backdropAnimationClasses = closedClass;
+          this.$forceUpdate();
+          this.$emit(chiDrawerHiddenEvent);
+        },
+        ANIMATION_DURATION.MEDIUM
+      );
+    }
+
+    this.$emit(chiDrawerHideEvent);
+  }
+
+  _documentClickHandler = (ev: Event): void => {
+    const drawerElement = this.$refs.drawerElement as HTMLElement;
+    const clickTarget = ev.target as HTMLElement;
+
+    if (!this.preventAutoHide) {
```

Comment:
I'd nest these 3 ifs

Reply:
Yes, I think I said it wrong, sorry. I meant to use the same if for all the checks

Reply:
```
if (!this.preventAutoHide &&
  drawerElement.classList.contains(activeClass)
  && !contains(drawerElement, clickTarget)) {
    this.hide();
}
```
This is what you mean guys, right?